controller/node_controller.go

```diff
@@ -941,90 +1028,88 @@ func (nc *NodeController) syncInstanceManagers(node *longhorn.Node) error {
 
 	log := getLoggerForNode(nc.logger, node)
 
-	imTypes := []longhorn.InstanceManagerType{
-		longhorn.InstanceManagerTypeAllInOne,
-		longhorn.InstanceManagerTypeEngine,
+	imTypes := getInstanceManagerTypes(node)
+	backendStoreDrivers := nc.getBackendStoreDrivers(node)
+
+	// Clean up instance manager for v2 data engine if the setting is disabled
+	if _, ok := backendStoreDrivers[longhorn.InstanceManagerTypeAllInOne][longhorn.BackendStoreDriverTypeV2]; !ok {
+		if err := nc.cleanupInstanceManagerForV2DataEngine(node); err != nil {
```

Comment:
Just instanceManager sync operations together. Any potential issue?

Reply:
In my opinion, it would make more sense for the setting controller or the instance-manager controller to handle the cleanup when the Setting CR is updated. The node controller doesn't appear to have a direct link to the instance-manager pods or the v2 engine setting, which makes the implementation less straightforward.

Reply:
Understand. Addressed it. Thank you.